pkg/ignition/ignition_test.go

```diff
@@ -0,0 +1,136 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2017 Red Hat, Inc.
+ *
+ */
+
+package ignition
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"os/user"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+)
+
+var _ = Describe("Ignition", func() {
+
+	tmpDir, _ := ioutil.TempDir("", "ignitiontest")
+
+	_, err := user.Current()
+	if err != nil {
+		panic(err)
+	}
+	dataCreationFunc := func(vmi *v1.VirtualMachineInstance, namespace string) error {
+		// fake creating the data file
+		_, err := os.Create("data.ign")
+
+		return err
+	}
+
+	BeforeSuite(func() {
+		err := SetIgnitionLocalDirectory(tmpDir)
+		if err != nil {
+			panic(err)
+		}
+		SetIgnitionDataCreationFunction(dataCreationFunc)
+	})
+
+	BeforeEach(func() {
+		SetIgnitionDataCreationFunction(dataCreationFunc)
+	})
+
+	AfterSuite(func() {
+		os.RemoveAll(tmpDir)
+	})
+
+	Describe("No Ignition data source", func() {
+		Context("when Ignition Data generation fails", func() {
+			It("should fail local data generation", func() {
+
+				timedOut := false
+				customCreationFunc := func(vmi *v1.VirtualMachineInstance, namespace string) error {
+					var args []string
```

Comment:
what is the purpose of this function? Just returning an error after a second? That can be done easier, e.g.
```
time.Sleep(time.Second)
return fmt.Errorf("timout")
```